### PR TITLE
Update CLI argument parsing

### PR DESCRIPTION
**Pull request by GDYendell**
* Update for compression now being an int
    * Add output-dir for where to save HDF5 files
    * Add more thorough checks for no-client mode
    * Rearrange argument parsing for clearer help message
    * Some other small changes for clarity

### CODE REVIEW

File: frameProcessor/src/FrameProcessorApp.cpp [hunk 1]
```diff
@@ -55,44 +55,47 @@ void parse_arguments(int argc, char** argv, po::variables_map& vm, LoggerPtr& lo
     config.add_options()
         ("logconfig,l",   po::value<string>(),
            "Set the log4cxx logging configuration file")
-        ("no-client,N",   po::bool_switch(),
-           "Enable full initial configuration to run without any client controller")
-        ("single-shot,S", po::bool_switch(),
-           "Shutdown after one dataset written")
+        ("ctrl",          po::value<std::string>()->default_value("tcp://127.0.0.1:5004"),
+         "Set the name of the shared memory frame buffer")
```
Comment (ajgdls) [inline frameProcessor/src/FrameProcessorApp.cpp]: Hi @GDYendell are you sure this comment relates to this command line parameter?  I've been confused by this, unless I misunderstand the diff.

File: frameProcessor/src/FrameProcessorApp.cpp [hunk 2]
```diff
@@ -55,44 +55,47 @@ void parse_arguments(int argc, char** argv, po::variables_map& vm, LoggerPtr& lo
...
+        ("sharedbuf",     po::value<std::string>()->default_value("FrameReceiverBuffer"),
+         "Set the control endpoint")
```
Comment (ajgdls) [inline frameProcessor/src/FrameProcessorApp.cpp]: Same comment, it looks like these might be switched actually.

Comment (GDYendell) [inline frameProcessor/src/FrameProcessorApp.cpp]: Interesting...

**Comment by GDYendell** [comment]
It should stop crashes for the specific example you gave me. If there are any others they should probably be fixed here before merging.

**Comment by ajgdls** [comment]
Oh I see, you check for all the items required before going into the no client mode.

**Pull request merged**